hack/update-pkg-manifest.sh

```diff
@@ -13,82 +13,91 @@
-cat ${DEPLOY_PATH}/${RH_BUNDLE_VERSION}/cockroach-operator.clusterserviceversion.yaml | sed "s+RH_COCKROACH_OP_IMAGE_PLACEHOLDER+${RH_COCKROACH_OP_IMG}+g; s+CREATED_AT_PLACEHOLDER+"$(date +"%FT%H:%M:%SZ")"+g"> ${DEPLOY_PATH}/${RH_BUNDLE_VERSION}/csv.yaml
-for v in $("$faq" -r '.CrdbVersions' "${REPO_ROOT}/crdb-versions.yaml" | cut -d ' ' -f2)
-do
-  vrs=${v//./_}
-  ENV_VAR_PLACEHOLDER="RH_COCKROACH_DB_IMAGE_PLACEHOLDER_${vrs}"
-  RH_COCKROACH_DATABASE_IMAGE_VERSION="registry.connect.redhat.com/cockroachdb/cockroach:${v}"
-  sed -i '' -e "s+${ENV_VAR_PLACEHOLDER}+${RH_COCKROACH_DATABASE_IMAGE_VERSION}+g" ${DEPLOY_PATH}/${RH_BUNDLE_VERSION}/csv.yaml
-done
-cd  ${DEPLOY_PATH}/${RH_BUNDLE_VERSION} && "$faq" -f yaml -o yaml --slurp '.[0].spec.install.spec.clusterPermissions+= [{serviceAccountName: .[2].metadata.name, rules: .[1].rules }] | .[0]' csv.yaml cockroach-database-role_rbac.authorization.k8s.io_v1_clusterrole.yaml cockroach-database-sa_v1_serviceaccount.yaml > cockroach-operator.v${RH_BUNDLE_VERSION}.clusterserviceversion.yaml
-shopt -s extglob
-rm -v !("cockroach-operator.v${RH_BUNDLE_VERSION}.clusterserviceversion.yaml"|"crdb.cockroachlabs.com_crdbclusters.yaml")
-shopt -u extglob
-# This is needed after csv generation
-cd ${REPO_ROOT}
-FILE_NAMES=(
-  config/manifests/bases/cockroach-operator.clusterserviceversion.yaml \
-)
-for YAML in "${FILE_NAMES[@]}"
-do
-   :
-   cat "${REPO_ROOT}/hack/boilerplate/boilerplate.yaml.txt" "${REPO_ROOT}/${YAML}" > "${REPO_ROOT}/${YAML}.mod"
-   mv "${REPO_ROOT}/${YAML}.mod" "${REPO_ROOT}/${YAML}"
-done
+  fi
+}
+
+generate_package_manifest() {
+  # Generate CSV in config/manifests and add boilerplate back (lost when regenerating)
+  operator-sdk generate kustomize manifests -q --apis-dir apis
+  hack/boilerplaterize hack/boilerplate/boilerplate.yaml.txt config/manifests/**/*.yaml
+
+  # TODO: packagemanifests is deprecated and scheduled to be remove in 2.0.0
+  kustomize build config/manifests | operator-sdk generate packagemanifests -q \
+    --version "${1}" \
+    ${2} \
+    --output-dir "${3}" \
+    --input-dir="${3}" \
+    --verbose
+}
+
+generate_csv() {
+  # replace RH_COCKROACH_OP_IMAGE_PLACEHOLDER with the proper image and CREATED_AT_PLACEHOLDER with the current time
+  cat ${1}/cockroach-operator.clusterserviceversion.yaml | sed \
+    "s+RH_COCKROACH_OP_IMAGE_PLACEHOLDER+${2}+g; s+CREATED_AT_PLACEHOLDER+"$(date +"%FT%H:%M:%SZ")"+g" > ${1}/csv.yaml
+
+  # for each RH_COCKROACH_DB_IMAGE_PLACEHOLDER_* set to the corresponding connect image
+  local version env img
+  for v in $(faq -r '.CrdbVersions' crdb-versions.yaml | cut -d ' ' -f2); do
+    version=${v//./_}
+    env="RH_COCKROACH_DB_IMAGE_PLACEHOLDER_${version}"
+    img="registry.connect.redhat.com/cockroachdb/cockroach:${v}"
```

Comment:
Should we pass ni the registry value so that we can deploy this into one of our registries for testing?

Reply:
If we did that, we'd need to publish an image for every version no?